Resources/config/util.xml

```diff
@@ -10,6 +10,8 @@
 
         <service id="fos_user.util.user_manipulator" class="FOS\UserBundle\Util\UserManipulator">
             <argument type="service" id="fos_user.user_manager" />
+            <argument type="service" id="event_dispatcher" />
+            <argument type="service" id="service_container" />
```

Comment:
Totally agree with you, I do not like injecting the container either. 

But UserEvent requires a Request object. I copy those lines form FOS\UserBundle\Security\LoginManager::logInUser, so I thought this was the better way to do this, or better to keep bundle's uniformity.

Y make UserEvent constructor $request property optional, because when event is lauched from fos:user commands there is not a request.

I think, unless with commands no request object is passed, if you attack directly to fos_user.util.user_manipulator service when a request is present, it's interesting to a request object is passed, just in case it's needed.